src/porepy/models/verification_setups/terzaghi.py

```diff
@@ -0,0 +1,723 @@
+
+        # North side: Neumann
+        bc.is_neu[:, north] = True
+        bc.is_dir[:, north] = False
+
+        # South side: Dirichlet (already set thanks to inheritance)
+
+        return bc
+
+    def time_dependent_bc_values_mechanics(
+        self, subdomains: list[pp.Grid]
+    ) -> np.ndarray:
+        """Boundary values for mechanics.
+
+        Parameters:
+            subdomains: List of subdomains on which to define boundary conditions.
+
+        Returns:
+            Array of boundary values. Only non-zero values are the ones associated to
+              the North side of the domain.
+
+        """
+        sd = subdomains[0]
+        vertical_load = self.params["vertical_load"]
+        _, _, _, north, *_ = self.domain_boundary_sides(sd)
+        bc_values = np.array([np.zeros(sd.num_faces), np.zeros(sd.num_faces)])
+        bc_values[1, north] = -vertical_load * sd.face_areas[north]
+        return bc_values.ravel("F")
+
+
+class ModifiedBoundaryConditionsSinglePhaseFlow(
+    mass.BoundaryConditionsSinglePhaseFlow,
+):
+
+    domain_boundary_sides: Callable[
+        [pp.Grid],
+        tuple[
+            np.ndarray,
+            np.ndarray,
+            np.ndarray,
+            np.ndarray,
+            np.ndarray,
+            np.ndarray,
+            np.ndarray,
+        ],
+    ]
+    """Utility function to access the domain boundary sides."""
+
+    def bc_type_darcy(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Define boundary condition types for the flow subproblem.
+
+        Parameters:
+            sd: Subdomain grid.
+
+        Returns:
+            Scalar boundary condition representation. All sides no flow, except the
+              North side which is set to a constant pressure.
+
+        """
+        # Define boundary regions
+        all_bf, _, _, north, *_ = self.domain_boundary_sides(sd)
+        north_bc = np.isin(all_bf, np.where(north)).nonzero()
+
+        # All sides Neumann, except the North which is Dirichlet
+        bc_type = np.asarray(all_bf.size * ["neu"])
+        bc_type[north_bc] = "dir"
+
+        bc = pp.BoundaryCondition(sd, faces=all_bf, cond=list(bc_type))
+
+        return bc
+
+
+class ModifiedPoromechanicsBoundaryConditions(
+    ModifiedBoundaryConditionsSinglePhaseFlow,
+    ModifiedBoundaryConditionsMechanicsTimeDependent,
+):
+    """Mixer class for poromechanics boundary conditions."""
+
+
+# -----> Solution strategy
+class ModifiedSolutionStrategy(
+    poromechanics.SolutionStrategyPoromechanics,
+):
+    results: list[StoreResults]
+    """List of store results objects."""
+
+    exact_sol: ExactSolution
+    """Exact solution object."""
+
+    plot_results: Callable
+    """Method that plots the presssure and degree of consolidation."""
+
+    def __init__(self, params: dict) -> None:
+        """Constructor of the class.
+
+        Parameters:
+            params: Parameters of the verification setup.
+
+        """
+        # Exact solution
```

Comment:
In general, I'm not a huge fan of too long inits. In particular, I think specific parameter values as a rule of thumb are not part of the solution strategy, but belong somehow outside this class.

Reply:
Point taken, but I could not find an alternative way to do this. The one path I explored was to subclass the `FluidConstants` and `SolidConstants` to re-define default values. But it was quite messy and added unnecessary complexity to the verification setup. Perhaps you have ideas on how to achieve this without in a more efficient way.

Regarding the parameters that are not material constants, I will use params.get(key, default_val) instead.

Reply:
Last sentence, done.

Reply:
You could define terzaghi_solid_values = {"foo": 42, "bar": 4.2, ...} and use them wherever you need them.

Reply:
Material parameters for the verification setup are now declared outside the mixer class.